src/interactivity/feature.js

```diff
@@ -72,34 +73,52 @@ import FeatureVizProperty from './featureVizProperty';
  * @property {FeatureVizProperty} width
  * @property {FeatureVizProperty} colorStroke
  * @property {FeatureVizProperty} widthStroke
+ * @property {FeatureVizProperty} symbol
+ * @property {FeatureVizProperty} symbolPlacement
  * @property {FeatureVizProperty[]} variables - Declared variables in the viz object
  * @property {function} reset - Reset custom feature vizs by fading out `duration` milliseconds, where `duration` is the first parameter to reset
  * @api
  */
+
 export default class Feature {
     constructor (rawFeature, viz, customizedFeatures, trackFeatureViz, idProperty) {
-        const variables = {};
-        Object.keys(viz.variables).map(varName => {
-            variables[varName] = new FeatureVizProperty(`__cartovl_variable_${varName}`, rawFeature, viz, customizedFeatures, trackFeatureViz, idProperty);
+        this.id = rawFeature[idProperty];
+
+        this._rawFeature = rawFeature;
+        this._featureVizParams = { rawFeature, viz, customizedFeatures, trackFeatureViz, idProperty };
+
+        this._defineFeatureVizProperties();
+        this._defineVariables(viz.variables);
+    }
+
+    _defineFeatureVizProperties () {
+        FEATURE_VIZ_PROPERTIES.forEach((property) => {
+            this[property] = this._buildFeatureVizProperty(property);
         });
+    }
 
-        this.id = rawFeature[idProperty];
-        this.color = new FeatureVizProperty('color', rawFeature, viz, customizedFeatures, trackFeatureViz, idProperty);
-        this.width = new FeatureVizProperty('width', rawFeature, viz, customizedFeatures, trackFeatureViz, idProperty);
-        this.strokeColor = new FeatureVizProperty('strokeColor', rawFeature, viz, customizedFeatures, trackFeatureViz, idProperty);
-        this.strokeWidth = new FeatureVizProperty('strokeWidth', rawFeature, viz, customizedFeatures, trackFeatureViz, idProperty);
-        this.symbol = new FeatureVizProperty('symbol', rawFeature, viz, customizedFeatures, trackFeatureViz, idProperty);
-        this.symbolPlacement = new FeatureVizProperty('symbolPlacement', rawFeature, viz, customizedFeatures, trackFeatureViz, idProperty);
+    _buildFeatureVizProperty (name) {
+        const { rawFeature, viz, customizedFeatures, trackFeatureViz, idProperty } = this._featureVizParams;
+        return new FeatureVizProperty(name, rawFeature, viz, customizedFeatures, trackFeatureViz, idProperty);
+    }
+
+    _defineVariables (vizVariables) {
+        const variables = {};
+        Object.keys(vizVariables).map(varName => {
+            const name = `__cartovl_variable_${varName}`;
+            variables[varName] = this._buildFeatureVizProperty(name);
+        });
         this.variables = variables;
     }
 
+    get rawFeature () {
+        return this._rawFeature;
```

Comment:
I don't know if we want to expose this. If the user doesn't use a property anywhere, then the property won't be here anyway.

If the user uses the property somewhere but not on the `viewportFeatures` call then the function is no longer pure, it is returning values for properties that are unknown for the function, i.e.: CVL magic is exposed in the open.

Reply:
Yeah, I had some doubts on this. Would you prefer just a `feature._rawFeature` call from outside?. I thought this might be useful in some use cases, like preparing popups, grids... but I don't have a strong opinion

Reply:
It is definitely useful, and if we expose it users will probably use it. However, I think that the problems outweigh the advantages.

Reply:
Then let's remove it!